Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -25,12 +27,73 @@ namespace ROS2
     URDFPrefabMaker::URDFPrefabMaker(const AZStd::string& modelFilePath, urdf::ModelInterfaceSharedPtr model, AZStd::string prefabPath)
         : m_model(model)
         , m_visualsMaker(modelFilePath, model->materials_)
+        , m_collidersMaker(new CollidersMaker(modelFilePath))
+        , m_stopBuildFlag(false)
         , m_prefabPath(std::move(prefabPath))
     {
         AZ_Assert(!m_prefabPath.empty(), "Prefab path is empty");
         AZ_Assert(m_model, "Model is nullptr");
     }
 
+    URDFPrefabMaker::~URDFPrefabMaker()
+    {
+        m_stopBuildFlag = true;
+        m_buildThread.join();
+    }
+
+    void URDFPrefabMaker::LoadURDF(BuildReadyCallback buildReadyCb)
+    {
+        m_notifyBuildReadyCb = buildReadyCb;
+
+        // Request the build of collider meshes by constructing .assetinfo files.
+        BuildAssetsForLink(m_model->root_link_);
+
+        // Wait for all collider meshes to be ready
+        m_buildThread = AZStd::thread(
+            [&]()
+            {
+                {
+                    AZ_Printf("CollisionMaker", "Waiting for URDF assets...");
+                    while (!m_collidersMaker->m_meshesToBuild.empty() && !m_stopBuildFlag)
+                    {
+                        {
+                            AZStd::lock_guard lock{ m_collidersMaker->m_buildMutex };
+                            for (auto iter = m_collidersMaker->m_meshesToBuild.begin(); iter != m_collidersMaker->m_meshesToBuild.end();
+                                 iter++)
+                            {
+                                AZ::Data::AssetId assetId;
+                                AZ::Data::AssetType assetType = AZ::AzTypeInfo<PhysX::Pipeline::MeshAsset>::Uuid();
+                                AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                                    assetId, &AZ::Data::AssetCatalogRequests::GetAssetIdByPath, iter->c_str(), assetType, false);
+                                if (assetId.IsValid())
+                                {
+                                    AZ_Printf("CollisionMaker", "Asset %s found and valid...", iter->c_str());
+                                    m_collidersMaker->m_meshesToBuild.erase(iter--);
+                                }
+                            }
+                        }
+                        if (!m_collidersMaker->m_meshesToBuild.empty())
+                        {
+                            AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(1000));
```

Comment:
Too much time to react to new things populating the queue - 50 ms is enough since we are only checking queue emptiness and a flag if there is no work.

Reply:
I changed the time to 50. The issue that will trace the threading improvement is would be #151